pkg/controller/snapshot_controller.go

```diff
@@ -844,11 +844,40 @@ func (ctrl *csiSnapshotController) getVolumeFromVolumeSnapshot(snapshot *crdv1.V
 		return nil, fmt.Errorf("failed to retrieve PV %s from the API server: %q", pvName, err)
 	}
 
+	// Verify binding between PV/PVC is still valid
+	bound := ctrl.IsVolumeBoundToClaim(pv, pvc)
+	if bound == false {
+		glog.Warningf("binding between PV %s and PVC %s is broken", pvName, pvc.Name)
+		return nil, fmt.Errorf("claim in dataSource not bound or invalid")
+	}
+
+	// Verify driver for PVC is the same as driver for VolumeSnapshot
+	if pv.Spec.PersistentVolumeSource.CSI == nil || pv.Spec.PersistentVolumeSource.CSI.Driver != ctrl.snapshotterName {
```

Comment:
Did we allow snapshotter name to be different from driver name in 1.0.x? If so, then we probably can't check this

Reply:
"snapshotter" is an input option, however, we do check in the controller to make sure it is the same as the driver name:

https://github.com/kubernetes-csi/external-snapshotter/blob/release-1.0/pkg/controller/snapshot_controller_base.go#L318